app/src/main/java/com/example/lunabeeusers/ui/overview/UserOverviewViewModel.kt

```diff
@@ -67,6 +92,32 @@ class UserOverviewViewModel @ViewModelInject constructor(
         }
     }
 
+    fun getUsersPageFromApi() {
+        viewModelScope.launch {
+            try {
+                _statut.value = Statut.LOADING
+                var listResult = repository.getUsersPage(pageToLoad)
+                _usersList += listResult
+                _statut.value = Statut.SUCCESS
+            } catch (t: Throwable) {
+                _statut.value = Statut.ERROR
+                t.printStackTrace()
+            }
+        }
+    }
+
+    fun loadNextPage() {
+        pageToLoad += 1
```

Comment:
Will be hard to keep it bug-free this way. For example, if the loading fails, you'll skip a page.
You should use the repository to know how many items you have.